### PR TITLE
Update to using direct from maven dependencies where possible

### PR DESCRIPTION
**Pull request by jonahgraham**
This is the next step in modernization and moving on from Orbit.

All the bundles that have fully OSGi bundles in maven are included in this update.

This PR contains the other update needed for 2023-06 M2

Part of #320

### CODE REVIEW

**Comment by ruspl-afed** [comment]
Thank you for working on this @jonahgraham 
Do you plan to change them all in one PR?

**Comment by jonahgraham** [comment]
> Do you plan to change them all in one PR?

Yes. I need to figure out how to document the updated steps now - but we have always updated all the dependencies in one PR, it just used to be by changing the Orbit version. As most of the versions in the target file were `"0.0.0"` the dependency version would change automagically.

**Comment by jonahgraham** [comment]
I don't understand why license check is failing here for icu, it passes in platform for AFAICT the same content.

**Comment by jonahgraham** [comment]
I don't see how it is passing for Eclipse Platform (e.g.  https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/actions/runs/4846605691/jobs/8636246851) - because I don't see license vetting having been done for 73.1 in iplab. 

The 72.1 is there in https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4354

We aren't releasing for a couple of months still, so this issue on its own isn't going to hold up merging this. But I would like to understand the flow of what is going wrong with this new check added in #386. Until now I have run license tool and manually verified the not automatically verified parts (e.g. https://ci.eclipse.org/cdt/job/run-dash-licenses/job/cdt_11_1/1/ for 11.1)

**Comment by jonahgraham** [comment]
Before this can be merged we also need the PGP signing which has a dependency on [EF Helpdesk #3070](https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3070)

File: pom.xml [hunk 1]
```diff
@@ -990,6 +990,22 @@
...
+				<artifactId>tycho-gpg-plugin</artifactId>
+				<version>${tycho-version}</version>
+				<executions>
+					<execution>
+						<id>pgpsigner</id>
+						<goals>
+						<goal>sign-p2-artifacts</goal>
+						</goals>
+						<configuration>
+							<keyname>4F23165B6AC51B15</keyname>
```
Comment (ruspl-afed) [inline pom.xml:1004]: Will this magic work for other Eclipse projects as well or this key is individual for every project @jonahgraham ?

Comment (jonahgraham) [reply pom.xml:1004]: It is specific to CDT's key. See https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3070

**Comment by jonahgraham** [comment]
Current state of this branch (maven-deps) is that I cannot sign because of `gpg: signing failed: No pinentry` error which according to https://wiki.eclipse.org/Jenkins#Common_issues_with_GPG indicates pinentry is not installed. But as pinentry is installed it is probably a custom image problem.

**Comment by jonahgraham** [comment]
> custom image problem.

So I am working on refreshing our custom images, but for 2023-06 I won't get this integrated.

**Pull request merged**